libcontainer/network_linux.go

```diff
@@ -98,3 +103,113 @@ func (l *loopback) attach(n *configs.Network) (err error) {
 func (l *loopback) detach(n *configs.Network) (err error) {
 	return nil
 }
+
+// devChangeNetNamespace allows to change the device name from a network namespace and optionally replace the existing name.
+// This function ensures that the move and rename operations occur simultaneously.
+// It preserves existing interface attributes, including IP addresses.
+func devChangeNetNamespace(name string, nsPath string, device configs.LinuxNetDevice) error {
+	logrus.Debugf("attaching network device %s with attrs %+v to network namespace %s", name, device, nsPath)
+	link, err := netlink.LinkByName(name)
+	if err != nil {
+		return fmt.Errorf("link not found for interface %s on runtime namespace: %w", name, err)
+	}
+
+	// set the interface down to change the attributes safely
```

Comment:
Why is this needed? It might not be obvious to everyone (it is not obvious to me, for example), so a comment explaining why would be great.

Or is it just in case the interface is being used, to make sure it is no longer used?

Reply:
expanded in a comment, fi the interface is being used then during the change of namespace there can be races, think a working interface has associated multiple properties and some of them dynamic or used by the system like ip neighbours cache, routes, ...